### PR TITLE
internal/tracer: use OpenTelemetry to do native Jaeger export

### PR DESCRIPTION
**Pull request by bobheadxi**
Since we can provide a Jaeger exporter directly to the OpenTelemetry library, this allows us to provide capabilities similar to before while removing our dependency on OpenTracing/Jaeger, allowing us to streamline `internal/trace` considerably (https://github.com/sourcegraph/sourcegraph/pull/40710) and allows the codebase to freely use OpenTracing _or_ OpenTelemetry without worrying about compatibility (e.g. without this change, OpenTelemetry instrumentation like #39300 would not export via `"observability.tracing": { "type": "jaeger" }"`.

This likely negates any real need for the original intent of https://github.com/sourcegraph/sourcegraph/issues/39399, which was to remove the Jaeger exporter option, because working with both libraries is now less of a concern, so I've adjusted the scope of that issue accordingly.

## Test plan

```
  "observability.tracing": {
    "sampling": "selective",
    "type": "jaeger",
    "debug": false
  },
```

then `sg start` and `sg run jaeger` (without opentelemetry collector running)

Then `&trace=1`:

image

### CODE REVIEW

File: internal/tracer/tracer.go [hunk 1]
```diff
@@ -175,19 +183,21 @@ func initTracer(logger log.Logger, opts *options, c conftypes.WatchableSiteConfi
...
-
+		exporter, err = exporters.NewJaegerExporter()
 	case OpenTelemetry:
-		return newOTelBridgeTracer(logger, opts)
+		exporter, err = exporters.NewOTelCollectorExporter(context.Background(), logger)
+	}
 
-	default:
-		return opentracing.NoopTracer{}, nil, nil, nil
+	if err != nil || exporter == nil {
```
Comment (burmudar) [inline internal/tracer/tracer.go:199]: Should we log the error and announce that we're using the NoopTracer ?

**Review by burmudar — APPROVED** [review]
One minor comment but other LGTM 🥳

**Review by sanderginn — APPROVED** [review]
Neat stuff.

File: internal/tracer/internal/exporters/jaeger.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	oteltracesdk "go.opentelemetry.io/otel/sdk/trace"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// NewJaegerExporter exports spans to a Jaeger collector or agent.
+func NewJaegerExporter() (oteltracesdk.SpanExporter, error) {
+	// Set configuration from jaegercfg package, to try and preserve back-compat with
+	// existing behaviour.
+	cfg, err := jaegercfg.FromEnv()
```
Comment (bobheadxi) [inline internal/tracer/internal/exporters/jaeger.go:17]: Thought - we lose Jaeger application-side sampling capabilities here, but I think it's okay for now given the next release is 4.0

**Pull request merged**